### ISSUE TITLE
Port twisted.names.common to Python 3

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6059]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#6059|
|Type|enhancement|
|Created|2012-10-04 18:12:21Z|
|Branch|https://github.com/twisted/twisted/tree/dns-common-py3-6059|

This is needed by `twisted.names.client`.

<details><summary>Searchable metadata</summary>

```
trac-id__6059 6059
type__enhancement enhancement
reporter__exarkun exarkun
priority__normal normal
milestone__Python_3_3_Minimal Python 3.3 Minimal
branch__branches_dns_common_py3_6059 branches/dns-common-py3-6059
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__names names
keywords__ 
time__1349374341000000 1349374341000000
changetime__1349817597000000 1349817597000000
version__None None
owner__exarkun exarkun

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

(In [36015]) Branching to 'dns-common-py3-6059'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun set owner to @itamarst|
|-|-|

Okay, there's some stuff.  Pretty straight-forward.  This ports `twisted.names.error` as well, since that ended up not needing much porting and has no dedicated test module of its own.  The test coverage here isn't great, but other parts of Twisted Names test suite exercise this code somewhat more, so when those are ported the coverage will improve.  I think maybe that's an okay approach for this code.  [Build results](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/dns-common-py3-6059)

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_3"></a>|@itamarst set owner to @exarkun|
|-|-|

  1. Two news files.
  1. You should verify in the tests that the result of ResolverBase.query is the result of the underlying function being called.
  1. `extractRecords` only has very implicit test coverage at best, it seems. At the very least you should open a ticket for that, if it's not needed for the use cases we're covering.

Other than that looks good, please merge.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun set status to `closed`|
|-|-|

(In [36039]) Merge dns-common-py3-6059

Author: exarkun
Reviewer: itamarst
Fixes: [#6058](https://github.com/twisted/twisted/issues/6058)
Fixes: [#6059](https://github.com/twisted/twisted/issues/6059)

Port `twisted.names.error` (trivial) and `twisted.names.common` to Python 3.
Also improve some of the test coverage and fix a few structural documentation
errors.
